packages/opentelemetry-propagation-utils/src/pubsub-propagation.ts

```diff
@@ -50,15 +50,26 @@ const patchArrayFilter = (
     return newArray;
   };
 
-  Object.defineProperty(messages, 'filter', {
+  Object.defineProperty(messages, "filter", {
     enumerable: false,
     value: patchedFunc,
   });
 };
 
+function isPromise(value: unknown): value is Promise<unknown> {
```

Comment:
Shouldn't the return type be `boolean` instead of `value is Promise<unknown>`

Reply:
it depends. the `value is` lets TypeScript know that if this predicate passes it can treat the value as a Promise.

```ts
if (isPromise(value)) {
  // TypeScript will know that inside this if block the value can be considered a promise for typechecking purposes
  value.then(x => ...) 
}
```

If `boolean` is used as a return value TypeScript will not narrow the type definition.

Reply:
Cool! I learned something new :)